pkg/collector/corechecks/containers/generic/processor.go

```diff
@@ -192,6 +192,11 @@ func (p *Processor) processContainer(sender sender.Sender, tags []string, contai
 		p.sendMetric(sender.Gauge, "container.pid.thread_limit", containerStats.PID.ThreadLimit, tags)
 	}
 
+	restartCount := float64(container.RestartCount)
```

Comment:
nit: You don't need to cast before comparing, only if `> 0` and you can use `pointer.Ptr(float64(container.RestartCount))` directly